Fix JSON parsing in tasks when '{' is in the GitVersion output

SimplyAName opened the pull request:
I have not contributed here before, so please let me know if there's anything I need to do/fix.
I've done my best to match existing styling and have included some unit tests for my change.

TLDR:
- Improved JSON parsing for GitVersion output
- Add new unit tests for the above change
- Added debug messages
- Updated README with basic instructions on how to get set up


This PR is mainly to fix #1711. I've been running into the same issue on my own pipelines and have tracked down the root cause to output parsing in the tasks.

The error occurs when a '{' character exists anywhere in the GitVersion output object, such as in a branch name. This caused the old parsing function to get an invalid JSON object, as the last '{' didn't belong to the JSON object, just the property string.

To fix this, I've added a function that will start with the old search (last '{' and '}') but then expand to the next '{' character and check if the JSON object is valid again. This continues till a valid object is found or sets the task to failed if it is unable to find one.

I've also included a new section on getting set up with this project in the README. I faced some issues setting it up for the first time, so I included some of the basic things I was missing. If the README changes should be in a separate PR, let me know

arturcic commented:
@SimplyAName please use rebase onto `main` instead of merging main into your branch. We prefer clean history with no intermediair merge commits

SimplyAName commented:
@arturcic I've rebased my branch and force-pushed to override the merge commits.
I also updated the README with the info about rebasing.

Lmk if anything needs fixing

arturcic reviewed (approved):
LGTM!

SimplyAName commented:
@arturcic Thanks for doing the fix!

Yeah, that sounds best long term. I'll look into that in future, thanks :)

I've rebased again after seeing some updates on the main branch. 
The SHA test seems quite flaky as it was having issues with the pipeline branch being out of date with main as well as rebase and force pushing.

Branch commit: 94569c821ccbd9359f38ad90421cbe530d28d742
Pipeline checkout commit: 5568eeac1f8f5dadb58144fc52b34b3b0b1c59a2

I've added a small update after the rebase to fix the commit hashes, but another rebase will cause it to fail again

Can this be merged soon so Dependabot doesn't cause me to rebase again

Thanks
